tools/build-toolchain.sh

```diff
@@ -34,139 +31,273 @@ export PATH="$PATH:$INSTALL_PATH/bin"
 JOBS="${JOBS:-$(getconf _NPROCESSORS_ONLN)}"
 JOBS="${JOBS:-1}" # If getconf returned nothing, default to 1
 
-# Additional GCC configure arguments
+# GCC configure arguments to use system GMP/MPC/MFPF
 GCC_CONFIGURE_ARGS=()
 
 # Dependency source libs (Versions)
 BINUTILS_V=2.39
 GCC_V=12.2.0
 NEWLIB_V=4.2.0.20211231
+GMP_V=6.2.0 
+MPC_V=1.2.1 
+MPFR_V=4.1.0
+MAKE_V=${MAKE_V:-""}
 
 # Check if a command-line tool is available: status 0 means "yes"; status 1 means "no"
 command_exists () {
-  (command -v "$1" >/dev/null 2>&1)
-  return $?
+    (command -v "$1" >/dev/null 2>&1)
+    return $?
 }
 
 # Download the file URL using wget or curl (depending on which is installed)
 download () {
-  if   command_exists wget ; then wget -c  "$1"
-  elif command_exists curl ; then curl -LO "$1"
-  else
-    echo "Install wget or curl to download toolchain sources" 1>&2
-    return 1
-  fi
+    if   command_exists wget ; then wget -c  "$1"
+    elif command_exists curl ; then curl -LO "$1"
```

Comment:
Why is this not using the "download" function??? as above...

Reply:
Not sure what you mean, what you mentioned *is* the implementation of the download function.